### DISCUSSION TITLE
THE .ENV PLEDGE

### DISCUSSION THREAD

**Discussion by patrick-streaming** [Cyfrin/moccasin-full-course-cu#2] [General]
# The Pledge
I solemnly swear, that I will never place a private key or secret phrase or mnemonic in a `.env` file that is associated with real funds. 

I will only place private keys in a `.env` file that have ONLY testnet ETH, LINK, or other cryptocurrencies. 

When I'm testing and developing, I will use a different wallet than the one associated with my real funds. 

I am aware that if I forget a `.gitignore` and push my key/phrase up to GitHub even for a split-second, or show my key/phrase to the internet for a split second, it should be considered compromised and I should move all my funds immediately. 

If I am unsure if my account has real funds in it, I will assume it has real funds in it. If I assume it has real funds in it, I will not use it for developing purposes. 

I am aware that even if I hit `add account` on my metamask (or other ETH wallet) I will get a new private key, but it will share the same secret phrase/mnemonic of the other accounts generated in that metamask (or other ETH wallet).

# Pledge Additions
For this course, I will only use funds associated with a brand new never-before-used metamask (or other ETH wallet). I am aware that my account associated with my private key is the same on testnets that it is on mainnets. 

If I must use a private key associated with real funds in the future, until I am 100% sure what I am doing I will always either:
1. Encrypt the private key MYSELF in such a way as [Patrick taught me](https://github.com/Cyfrin/web3py-favorites-cu/blob/main/encrypt_key.py)
2. Use the built-in functionality of [moccasin](https://github.com/cyfrin/moccasin) to encrypt my private keys
3. Use the command line way to pass private keys, and delete my command line history right after ONLY AS A LAST OPTION

If I never actually deploy anything to mainnet myself or work with a private key with real funds, I do not need to be concerned.

**Comment by patrick-streaming** [comment]
I WILL BE SAFE

**Reply by ridley88** [reply]
I WILL BE SAFE

**Reply by Dremas22** [reply]
> I WILL BE SAFE

I WILL BE SAVE

**Comment by EngrPips** [comment]
I WILL BE SAFE.

**Reply by Dremas22** [reply]
I WILL BE SAVE

**Comment by GoldTeaaa** [comment]
I WILL BE SAFE!!

**Comment by ghost** [comment]
I will be safe!

**Comment by GuireWire** [comment]
I WILL BE SAFE!!!!

**Comment by doble196** [comment]
# 🛡️ Vyper-Pilled Dev Safety Pledge 🛡️

## I, Rensley R., the First "Vyper-Pilled Dev," Solemnly Swear:

**1. I will never place a private key, secret phrase, or mnemonic in a `.env` file associated with real funds.**  

**2. I will never place private keys, even for _testnet_ ETH, LINK, or other cryptocurrencies, in plain text, because:**  
   - **Good habits in testing are great habits in production.**  

**3. I will always use a separate wallet for testing and development, isolated from my real funds wallet.**  

---

## 🧠 Trick Question 🤔

**Is it okay to keep private keys for testnet ETH in plain text just because it's testnet?**  
🚫 **NO!** Treat testnet keys like real keys because:  
- **Sloppy habits in testing lead to catastrophic mistakes in production.**  

---

## 🛑 Immediate Action if Keys Are Exposed

If I accidentally push my key or phrase to GitHub (even for a moment) or show it publicly:

1. I will **consider it compromised immediately**.
2. I will move all funds to a new wallet, whether it’s testnet or mainnet.

**When in doubt:**  
- If I suspect an account contains real funds, I will assume it does and avoid using it for development.

---

## 🔑 Understanding Wallet Behavior

- I understand that adding a new account in Metamask (or similar wallets) generates a new private key.  
- **All accounts under a single wallet share the same secret phrase/mnemonic.**

---

## 🔒 Secure Private Keys

If I must use a private key associated with **real funds** in the future:

1. I will **encrypt the private key myself** using best practices.
2. I will use **built-in encryption tools**, such as those provided by moccasin or another secure framework.
3. **As an absolute last resort**:
   - Pass private keys through the **command line**.  
   - **Delete the command line history** immediately afterward.

---

## 📜 Pledge Additions

For this course and all future projects:

1. I will use a **brand-new Metamask wallet** (or equivalent) with no prior usage.  
2. I understand that wallet addresses are **identical on testnets and mainnets**.

---

## 💡 Final Note

Even in testing, **never** keep private keys, secret phrases, or mnemonics in plain text. By building good habits today, I ensure safe deployments tomorrow.

---

**🚀 Let’s build securely, together.**  
🔗 **Project Repository**: [NFTeria](https://github.com/NFTeria)  
🐦 **Find me on Twitter (X)**: [@Rensley_Ramos](https://x.com/Rensley_Ramos)

---

> *Pledge signed by: __Rensley Ramos__* 🐍✨

**Comment by Torres91** [comment]
I WILL BE SAFE

**Comment by Baloo8721** [comment]
I WILL BE SAFE

**Comment by bycraft-betrue** [comment]
I WILL BE SAFE

**Comment by s3bc40** [comment]
I WILL BE SAFE

**Comment by Scofield-Idehen** [comment]
If you see this, please remember I won't be that moron and if I ever make a mistake may the seven gods forgive me not.

👍🏾

**Comment by pattool** [comment]
I WILL BE SAFE

**Comment by draju1980** [comment]
I WILL BE SAFE

**Comment by vycod** [comment]
I WILL BE SAFE

**Comment by Jon-Mervedel** [comment]
I WILL BE SAFE

**Comment by DiocletianIV** [comment]
I WILL BE SAFE

**Comment by didier-building** [comment]
I WILL BE SAFE!

**Comment by ghost** [comment]
I WILL BE SAFE

**Comment by Sir-Shaedy** [comment]
I WILL AND SHALL BE SAFE.

**Comment by Dev568145** [comment]
I WILL BE SAFE

**Comment by Bborde1** [comment]
I WILL BE SAFE

**Comment by Mikegor01** [comment]
I WILL BE SAFE !

**Comment by Victor-Okpukpan** [comment]
I WILL BE SAFE!

**Comment by okekekingsley** [comment]
I WILL BE SAFE

**Comment by Tariq-Imran** [comment]
I WILL BE SAFE

**Comment by arefxv** [comment]
I WILL BE SAFE

**Comment by Anand-7** [comment]
I WILL BE SAFE

**Comment by diegobianqui** [comment]
I WILL BE SAFE

**Comment by AshutoshShekharProjects** [comment]
I WILL BE SAFE!

**Comment by tranielmp8** [comment]
I WILL BE SAFE!

**Comment by abhaykumar129** [comment]
I WILL BE SAFE

**Comment by Jamie4141** [comment]
I WILL BE SAFE

**Comment by JFORNELLS** [comment]
I WILL BE SAFE!

**Comment by sirdoxa** [comment]
I WILL BE SAFE

**Comment by namedsajjad** [comment]
### I WILL BE SAFE!

**Comment by Abhinav-Malik-154** [comment]
I WILL BE SAFE

**Comment by sanbal13** [comment]
I WILL BE SAFE
